Limit number of profile pictures in the penny chat reminder

# Description
We received an error from sentry while trying to send our Penny Chat reminders for a recent chat with 10 participants. 
```
The request to the Slack API failed.
The server responded with: {'ok': False, 'error': 'invalid_blocks', 'response_metadata': {'messages': ['[ERROR] no more than 10 items allowed [json-pointer:/blocks/4/elements]']}}
```
After some digging, I found that we are creating a list of profile images to attach to the reminder message, along with a string for how many participants there are. Since we had 10 participants, these images plus the attending string caused the error. We need to limit the number of attendees we show profiles for, and then have an adjusted messaged like "and 3 more attending" if we go over the limit.

## Requirements
- [x] Limit the number of profiles pictures being sent with the reminder message
- [x] Update message if over the limit to "and X more attending"

--- a/bot/tasks/pennychat.py
+++ b/bot/tasks/pennychat.py
@@ -36,6 +36,8 @@ PENNY_CHAT_ID = 'penny_chat_id'
 PREVIEW, INVITE, UPDATE, REMIND = 'review', 'invite', 'update', 'remind'
 PENNY_CHAT_DETAILS_BLOCKS_MODES = {PREVIEW, INVITE, UPDATE, REMIND}
 
+MAX_PARTICIPANT_PROFILES = 9
+
 
 def background(*args, **kwargs):
     """This is just a silly hack because we don't have an easy way of shutting off async for tests.
@@ -274,6 +276,13 @@ def _penny_chat_details_blocks(penny_chat_invitation, mode=None):
         default = "https://avatars.slack-edge.com/2020-05-25/1144415980914_c8a4ff7c783de54f72e5_512.png"
         size = 40
         profiles = []
+
+        # issue-349: set max profile images to avoid too many blocks
+        profile_overage = 0
+        if len(participants) > MAX_PARTICIPANT_PROFILES:
+            profile_overage = len(participants) - MAX_PARTICIPANT_PROFILES
+            participants = participants[0:MAX_PARTICIPANT_PROFILES]
+
         for participant in participants:
             # get the (gravatar) profile image of a participant
             email = participant.email
@@ -292,7 +301,7 @@ def _penny_chat_details_blocks(penny_chat_invitation, mode=None):
                 'image_url': gravatar_url
             })
 
-        number_votes_text = f"{len(profiles)} attending"
+        profiles_text = f"{len(profiles)} attending" if profile_overage == 0 else f"& {profile_overage} more attending"
 
         attendee_images = [
             {
@@ -307,7 +316,7 @@ def _penny_chat_details_blocks(penny_chat_invitation, mode=None):
             {
                 'type': 'plain_text',
                 'emoji': True,
-                'text': number_votes_text
+                'text': profiles_text
             }
         ]
 

--- a/bot/tests/processors/test_pennychat.py
+++ b/bot/tests/processors/test_pennychat.py
@@ -5,10 +5,11 @@ import time
 import pytest
 from pytz import timezone, utc
 from sentry_sdk import capture_exception
-from common.tests.fakes import PennyChatSlackInvitationFactory
+from common.tests.fakes import PennyChatSlackInvitationFactory, UserFactory, SocialProfileFactory
 
 from bot.processors.pennychat import PennyChatBotModule
 import bot.processors.pennychat as penny_chat_constants
+from bot.tasks.pennychat import _penny_chat_details_blocks
 from pennychat.models import (
     PennyChatSlackInvitation,
     Participant,
@@ -361,3 +362,21 @@ def test_PennyChatBotModule_attendance_selection(
 
     # Make sure the other attendee wasn't affected
     assert Participant.objects.get(penny_chat=penny_chat, user=some_other_attendee.user).role == Participant.ATTENDEE
+
+
+@pytest.mark.django_db
+def test_penny_chat_reminders_blocks(mocker):
+    invite = PennyChatSlackInvitationFactory()
+    organizer = UserFactory()
+    organizer_profile = SocialProfileFactory()
+    Participant.objects.create(user=organizer, penny_chat=invite, role=Participant.ORGANIZER)
+    for i in range(12):
+        user = UserFactory()
+        Participant.objects.create(user=user, penny_chat=invite, role=Participant.ATTENDEE)
+
+    with mocker.patch('bot.tasks.pennychat.get_or_create_social_profile_from_slack_id', return_value=organizer_profile):
+        reminder_blocks = _penny_chat_details_blocks(invite, 'remind')
+
+    # Check that the correct number of blocks were created
+    assert len(reminder_blocks[4]['elements']) == 10
+    assert reminder_blocks[4]['elements'][-1]['text'] == '& 4 more attending'

